source/slang/slang-ir.cpp

```diff
@@ -7082,6 +7098,8 @@ namespace Slang
                 module->getDeduplicationContext()->getConstantMap().remove(IRConstantKey{ constInst });
             }
             module->getDeduplicationContext()->getInstReplacementMap().remove(this);
+            if (auto func = as<IRGlobalValueWithCode>(this))
+                module->invalidateDominatorTree(func);
```

Comment:
Could this be a more general `module->invalidateAnalysis(this)`?

Reply:
Done.